public/app/features/scenes/variables/variants/query/QueryVariable.tsx

```diff
@@ -125,6 +125,19 @@ export class QueryVariable extends MultiValueVariable<QueryVariableState> {
     );
   }
 
+  private async getDataSource(): Promise<DataSourceApi> {
+    const ds = this.state.datasource ?? '';
+    if (typeof ds === 'string') {
+      const interpolatedDS = sceneGraph.interpolate(this, ds);
+      if (Array.isArray(interpolatedDS) && interpolatedDS.length > 0) {
+        return getDataSourceSrv().get(String(interpolatedDS[0]));
+      }
+      return getDataSourceSrv().get(String(interpolatedDS));
+    }
+
+    return getDataSourceSrv().get();
+  }
```

Comment:
This should not be needed as getDataSourceSrv already interpolates it, so if you do. 

 ```
return getDataSourceSrv().get(this.state.datasource, { __sceneObject: { text: '__sceneObject', value: this }});
```

it should work

Reply:
Ups, I didn't realize, let me change it, thanks!